File: app/src/main/java/wangdaye/com/geometricweather/db/converters/WeatherSourceConverter.java
package wangdaye.com.geometricweather.db.converters;

import org.greenrobot.greendao.converter.PropertyConverter;

import wangdaye.com.geometricweather.basic.models.options.provider.WeatherSource;

public class WeatherSourceConverter implements PropertyConverter<WeatherSource, String> {

    @Override
    public WeatherSource convertToEntityProperty(String databaseValue) {
        return WeatherSource.valueOf(databaseValue);
    }

    @Override
    public String convertToDatabaseValue(WeatherSource entityProperty) {
        return entityProperty.name();
    }
}

File: app/src/main/java/wangdaye/com/geometricweather/db/converters/WeatherSourceConverter.kt
package wangdaye.com.geometricweather.db.converters

import org.greenrobot.greendao.converter.PropertyConverter
import wangdaye.com.geometricweather.common.basic.models.options.provider.WeatherSource

class WeatherSourceConverter : PropertyConverter<WeatherSource, String?> {

    override fun convertToEntityProperty(databaseValue: String?) =
        // use get instance method but not getValue method.
        WeatherSource.getInstance(databaseValue ?: "")

    override fun convertToDatabaseValue(entityProperty: WeatherSource) =
        entityProperty.id
}